hedera-node/src/main/java/com/hedera/services/state/migration/MigrationRecordsManager.java

```diff
@@ -124,6 +143,36 @@ private TransactionBody.Builder synthStakingFundCreate(final long autoRenewPerio
 		return TransactionBody.newBuilder().setCryptoCreateAccount(txnBody);
 	}
 
+	private void publishContractFreeAutoRenewalRecords(final Instant now) {
+		final var changedContracts = new HashMap<EntityNum, Long>();
+		accounts.get().forEach((a, b) -> {
+			if (b.isSmartContract()) {
+				changedContracts.put(a, b.getExpiry());
+			}
+		});
+
+		for (var entry : changedContracts.entrySet()) {
+			final var contractNum = entry.getKey().toEntityId();
+			final var newExpiry = entry.getValue();
+
+			final var receipt = new TxnReceipt();
+			receipt.setAccountId(contractNum);
+
+			final var synthBody = syntheticTxnFactory.synthContractAutoRenew(contractNum.asNum(), newExpiry);
+			final var memo = String.format(CONTRACT_UPGRADE_MEMO, contractNum.num(), newExpiry);
+			final var txnId = new TxnId(contractNum, MISSING_INSTANT, false, USER_TRANSACTION_NONCE);
+			final var synthRecord = ExpirableTxnRecord.newBuilder()
+					.setTxnId(txnId)
+					.setMemo(memo)
+					.setReceipt(receipt)
+					.setConsensusTime(RichInstant.fromJava(now));
+
+			recordsHistorian.trackPrecedingChildRecord(DEFAULT_SOURCE_ID, synthBody, synthRecord);
+			sigImpactHistorian.markEntityChanged(contractNum.num());
```

Comment:
Since you changed all the actual entities in `migrate()`, you don't need this.

Reply:
Got it ! Fixed ! Thanks !